src/Javinizer/Public/Get-DmmData.ps1

```diff
@@ -45,7 +48,7 @@ function Get-DmmData {
         $movieDataObject = [PSCustomObject]@{
             Source        = if ($Url -match '/en/') { 'dmm' } else { 'dmmja' }
             Url           = $Url
-            Id            = Get-DmmId -WebRequest $webRequest
+            Id            = if ($IdPreference -eq "id") { Get-DmmId -WebRequest $webRequest } elseif ($IdPreference -eq "contentid") { Get-DmmContentId -WebRequest $webRequest }

```

Comment:
I did a bit of thinking about having the Id convert to ContentId directly on the scrapers themselves.

I think it would be better to change it only on the aggregated data object, since if people were using Javinizer solely for the scraper functions, they wouldn't want the original scraper output to be changed depending on an Id preference.

My approach would be to consolidate the settings into a single setting `scraper.option.idpreference`, remove `ContentId` from the aggregated data object, and set the `Id` during the foreach loop that assigns all the aggregated data fields.
```powershell
foreach ($field in $metadataFields) {
    $metadataPriority = (Get-Variable -Name "$($field)Priority" -ValueOnly)
    foreach ($priority in $metadataPriority) {
        $sourceData = $Data | Where-Object { $_.Source -eq $priority }

        if ($null -eq $aggregatedDataObject.$field) {
            if ($field -eq 'AlternateTitle') {
                $aggregatedDataObject.$field = $sourceData.Title
            } elseif ($field -eq 'Id') {
                if ($IdPreference -eq 'contentid') {
                    $aggregatedDataObject.$field = $sourceData.ContentId
                } else {
                    $aggregatedDataObject.$field = $sourcedata.Id
                }
            } else {
                $aggregatedDataObject.$field = $sourceData.$field
            }
            Write-JVLog -Write:$script:JVLogWrite -LogPath $script:JVLogPath -WriteLevel $script:JVLogWriteLevel -Level Debug -Message "[$($Data[0].Id)] [$($MyInvocation.MyCommand.Name)] [$field - $priority] Set to [$($sourceData.$field | ConvertTo-Json -Compress)]"
        }
    }
}
```

What do you think of this approach?

Reply:
I personally like this approach.

To be honest, I wasn't quite pleased with what I have done. I prefer not to have the logic encoded in the model (in this case `$movieDataObject`). I will include this in the changes.